safehttp/plugins/csp/csp_test.go

```diff
@@ -0,0 +1,226 @@
+	}{
+		{
+			name:        "No policies",
+			interceptor: Interceptor{},
+		},
+		{
+			name: "Default policy",
+			interceptor: func() Interceptor {
+				p := Default("https://foo.com/collector")
+				p.EnforcementPolicy.readRand = readRand
+				return p
+			}(),
+			wantEnforcementPolicy: []string{"object-src 'none'; script-src 'nonce-AAECAwQFBgc=' 'unsafe-inline' 'unsafe-eval' 'strict-dynamic' https: http:; base-uri 'none'; report-uri https://foo.com/collector"},
+		},
+		{
+			name: "Report",
+			interceptor: Interceptor{
+				ReportOnlyPolicy: &Policy{
+					Directives: []*PolicyDirective{
+						{Directive: DirectiveScriptSrc, Values: []string{ValueUnsafeInline}, AddNonce: false},
+					},
+				},
+			},
+			wantReportPolicy: []string{"script-src 'unsafe-inline'"},
+		},
+		{
+			name: "Report with nonces",
+			interceptor: Interceptor{
+				ReportOnlyPolicy: func() *Policy {
+					p := NewPolicy("https://foo.com/collector")
+					p.readRand = readRand
+					return p
+				}(),
+			},
+			wantReportPolicy: []string{"object-src 'none'; script-src 'nonce-AAECAwQFBgc=' 'unsafe-inline' 'unsafe-eval' 'strict-dynamic' https: http:; base-uri 'none'; report-uri https://foo.com/collector"},
+		},
+		{
+			name: "Report and enforce",
+			interceptor: Interceptor{
+				ReportOnlyPolicy: &Policy{
+					Directives: []*PolicyDirective{
+						{Directive: DirectiveScriptSrc, Values: []string{ValueUnsafeInline}, AddNonce: false},
+						{Directive: DirectiveBaseURI, Values: []string{ValueNone}, AddNonce: false},
+					},
+				},
+				EnforcementPolicy: &Policy{
+					Directives: []*PolicyDirective{
+						{Directive: DirectiveScriptSrc, Values: []string{ValueUnsafeInline}, AddNonce: false},
+					},
+				},
+			},
+			wantEnforcementPolicy: []string{"script-src 'unsafe-inline'"},
+			wantReportPolicy:      []string{"script-src 'unsafe-inline'; base-uri 'none'"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rr := safehttptest.NewResponseRecorder()
+			req := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+
+			tt.interceptor.Before(rr.ResponseWriter, req)
+
+			h := rr.Header()
+			if diff := cmp.Diff(tt.wantEnforcementPolicy, h.Values("Content-Security-Policy")); diff != "" {
+				t.Errorf("h.Values(\"Content-Security-Policy\") mismatch (-want +got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff(tt.wantReportPolicy, h.Values("Content-Security-Policy-Report-Only")); diff != "" {
+				t.Errorf("h.Values(\"Content-Security-Policy-Report-Only\") mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestAlreadyClaimed(t *testing.T) {
+	headers := []string{
+		"Content-Security-Policy",
+		"Content-Security-Policy-Report-Only",
+	}
+
+	for _, h := range headers {
+		t.Run(h, func(t *testing.T) {
+			rr := safehttptest.NewResponseRecorder()
+			if _, err := rr.ResponseWriter.Header().Claim(h); err != nil {
+				t.Fatalf("rr.ResponseWriter.Header().Claim(h) got: %v want: nil", err)
```

Comment:
If you have more than one return value, it's IMHO a good idea to write: `foo() got err: %v, want: nil` instead.